tests/test_requests.py

```diff
@@ -606,6 +621,115 @@ def test_unicode_header_name(self, httpbin):
     def test_pyopenssl_redirect(self, httpbin_secure, httpbin_ca_bundle):
         requests.get(httpbin_secure('status', '301'), verify=httpbin_ca_bundle)
 
+    def test_verify(self):
+        """verify=True works with requests.get"""
+        with pytest.raises(SSLError):
+            requests.get('https://testssl-expire-r2i2.disig.sk/index.en.html')
+
+        with pytest.raises(SSLError):
+            requests.get('https://testssl-expire-r2i2.disig.sk/index.en.html')
+
+        with pytest.raises(SSLError):
+            requests.get('https://self-signed.badssl.com/')
+
+        with pytest.raises(SSLError):
+            requests.get('https://self-signed.badssl.com/')
+
+        session = requests.Session()
+
+        with pytest.raises(SSLError):
+            session.get('https://testssl-expire-r2i2.disig.sk/index.en.html')
+
+        with pytest.raises(SSLError):
+            session.get('https://testssl-expire-r2i2.disig.sk/index.en.html')
+
+        with pytest.raises(SSLError):
+            session.get('https://self-signed.badssl.com/')
+
+        with pytest.raises(SSLError):
+            session.get('https://self-signed.badssl.com/')
```

Comment:
Requests is as much as possible moving to offline testing: that is, testing where we do not need to hit real URLs. That's because this kind of testing is flaky and requires a good network floating around. It would be better if we could create endpoints locally that exhibit similar problems.